crates/executor/src/executor/mod.rs

```diff
@@ -166,10 +162,9 @@ where
             }
 
             // Modify the transaction environment with the current transaction.
-            evm = evm
-                .modify()
-                .with_tx_env(Self::prepare_tx_env(&transaction, raw_transaction)?)
-                .build();
+            let reth_tx = TransactionSigned::decode_2718(&mut raw_transaction).map_err(ExecutorError::RLPError)?;
```

Comment:
@clabby This decoding is failing in the tests. Having trouble figuring out why. If you have any instincts, I'd love your input as I keep digging.

Reply:
Likely because some of these txs are deposits. I assume `TransactionSigned` may not support the deposit (0x7e) variant, given that no features are enabled for `reth-primitives`.